src/components/screens/editor/editor-screen.tsx

```diff
@@ -74,99 +70,55 @@ const EditorScreen: React.FC = () => {
           data: { ...newMetaform } as any
         }
       });
-      const currentPath = window.location.pathname;
       navigate(`${currentPath}/${newMetaform.slug}/${newMetaformVersion.id}`);
     } catch (e) {
-      errorContext.setError(strings.errorHandling.adminFormsScreen.listForms, e);
+      errorContext.setError(strings.errorHandling.adminFormsScreen.createForm, e);
     }
   };
 
   /**
-   * Renders MetaformVersions Listing
-   * 
-   * @param metaform metaform
+   * Deletes a Metaform or MetaformVersion
    */
-  const buildVersionRows = (metaform: Metaform) => {
-    const versions = metaformRows.find(x => x.metaform.id === metaform.id)?.versions;
-    if (!versions) {
-      return;
+  const deleteMetaformOrVersion = async () => {
+    const id = selectedId!;
```

Comment:
id should be checked for undefined and I suggest that you don't check it from state but pass it to this function as parameter. There's always change that state doesn't update fast enough for id to exist.